autolab/Autolab
# File manager exception fix

## Description
This PR fixes the routing when a user is not allowed to access a path. This also adds unit tests which tests this functionality for this fix.

Resolves #2228 

## How Has This Been Tested?
- Create a course and assign a student to that course
- Navigate to that course in the file manager and copy the link
- Sign in as a student using dev login and try to access the link
- Note that you will not be able to see anything in that path and note the flash[:error]
- Run the new unit tests: `rake spec SPEC=./spec/controllers/file_manager_controller_spec.rb` and make sure that everything passes

## Types of changes

## Checklist:
